src/components/views/dialogs/DevtoolsDialog.js

```diff
@@ -0,0 +1,294 @@
+            this.setState({ eventType: null });
+        } else {
+            this.props.onBack();
+        }
+    }
+
+    render() {
+        if (this.state.event) {
+            return <div className="mx_ViewSource">
+                <div className="mx_Dialog_content">
+                    <pre>{JSON.stringify(this.state.event, null, 2)}</pre>
+                </div>
+                <div className="mx_Dialog_buttons">
+                    <button onClick={this.onBack}>{ _t('Back') }</button>
+                </div>
+            </div>;
+        }
+
+        const rows = [];
+
+        if (this.state.eventType === null) {
+            Object.keys(this.roomStateEvents).forEach((evType) => {
+                const stateGroup = this.roomStateEvents[evType];
+                const stateKeys = Object.keys(stateGroup);
+
+                let onClickFn;
+                if (stateKeys.length > 1) {
+                    onClickFn = this.browseEventType(evType);
+                } else if (stateKeys.length === 1) {
+                    onClickFn = this.onViewSourceClick(stateGroup[stateKeys[0]]);
+                }
+
+                rows.push(<button key={evType} onClick={onClickFn}>{ evType }</button>);
+            });
+        } else {
+            const evType = this.state.eventType;
+            const stateGroup = this.roomStateEvents[evType];
+            Object.keys(stateGroup).forEach((stateKey) => {
+                const ev = stateGroup[stateKey];
+                rows.push(<button key={stateKey} onClick={this.onViewSourceClick(ev)}>{stateKey}</button>);
+            });
+        }
+
+        return <div>
+            <div className="mx_Dialog_content">
+                {rows}
+            </div>
+            <div className="mx_Dialog_buttons">
+                <button onClick={this.onBack}>{ _t('Back') }</button>
+            </div>
+        </div>;
+    }
+}
+
+export default class DevtoolsDialog extends React.Component {
+    static propTypes = {
+        roomId: React.PropTypes.string.isRequired,
+        onFinished: React.PropTypes.func.isRequired,
+    };
+
+    state = {
+        mode: null,
+    };
+
+    constructor(props, context) {
+        super(props, context);
+        this.onBack = this.onBack.bind(this);
+        this.onCancel = this.onCancel.bind(this);
+    }
+
+    componentWillUnmount() {
+        this._unmounted = true;
+    }
+
+    _setMode(mode) {
+        return () => {
+            this.setState({ mode });
+        };
+    }
+
+    onBack() {
+        this.setState({ mode: null });
+    }
+
+    onCancel() {
+        this.props.onFinished(false);
+    }
+
+    render() {
+        let body;
+
+        if (this.state.mode) {
+            body = <this.state.mode {...this.props} onBack={this.onBack} />;
+        } else {
+            body = <div>
+                <div className="mx_Dialog_content">
+                    <button onClick={this._setMode(SendCustomEvent)}>{ _t('Send Custom Event') }</button>
```

Comment:
I suppose we should be doing the whole skinning dance here.

Reply:
Skinning dance?

Reply:
Using the component skinning that we support. i.e. `sdk.getComponent('bla')`

Reply:
oh so you're suggesting that I pull in the 3 tools through the sdk? ooi why is that done if they're defined at the riot-web level which is designed to be the top level?